src/diffpy/utils/diffraction_objects.py

```diff
@@ -248,15 +248,27 @@ def _set_array_from_range(self, begin, end, step_size=None, n_steps=None):
             array = np.linspace(begin, end, n_steps)
         return array
 
-    def get_angle_index(self, angle):
-        count = 0
-        for i, target in enumerate(self.angles):
-            if angle == target:
+    def get_array_index(self, xtype, value):
+        """
+        returns the index of a given value in the array associated with the specified xtype
+
+        Parameters
+        ----------
+        xtype str
+            the xtype used to access the array
+        value float
+            the target value to search for
+
+        Returns
+        -------
+        the index of the value in the array
+        """
+        if self.on_xtype(xtype) is None:
+            raise ValueError(_xtype_wmsg(xtype))
```

Comment:
I think I can remove this if I let `on_xtype()` to raise an error for invalid xtypes. Will do that in the other PR.